backend/infrahub/core/account.py

```diff
@@ -145,24 +145,93 @@ async def query_init(self, db: InfrahubDatabase, **kwargs: Any) -> None:
+            WITH account_role, r1, r2, (r1.status = "active" AND r2.status = "active") AS is_active
+            ORDER BY account_role.uuid, r2.branch_level DESC, r2.from DESC, r1.branch_level DESC, r1.from DESC
+            WITH account_role, head(collect(is_active)) as latest_is_active
+            WHERE latest_is_active = TRUE
+            RETURN account_role
+        }
+        WITH account_role
+
+        CALL {
+            WITH account_role
+            MATCH (account_role)-[r1:IS_RELATED]->(:Relationship {name: "role__permissions"})<-[r2:IS_RELATED]-(object_permission:%(object_permission_node)s)
+            WHERE all(r IN [r1, r2] WHERE (%(branch_filter)s))
+            WITH object_permission, r1, r2, (r1.status = "active" AND r2.status = "active") AS is_active
+            ORDER BY object_permission.uuid, r2.branch_level DESC, r2.from DESC, r1.branch_level DESC, r1.from DESC
+            WITH object_permission, head(collect(is_active)) as latest_is_active
+            WHERE latest_is_active = TRUE
+            RETURN object_permission
+        }
+        WITH object_permission
+        CALL {
+            WITH object_permission
+            MATCH (object_permission)-[r1:HAS_ATTRIBUTE]->(:Attribute {name: "branch"})-[r2:HAS_VALUE]->(object_permission_branch:AttributeValue)
+            WHERE all(r IN [r1, r2] WHERE (%(branch_filter)s))
+            WITH object_permission_branch, r1, r2, (r1.status = "active" AND r2.status = "active") AS is_active
+            ORDER BY object_permission_branch.uuid, r2.branch_level DESC, r2.from DESC, r1.branch_level DESC, r1.from DESC
+            WITH object_permission_branch, head(collect(is_active)) as latest_is_active
+            WHERE latest_is_active = TRUE
+            RETURN object_permission_branch
+        }
         WITH object_permission, object_permission_branch
-        WHERE all(r IN relationships(group_path) WHERE (%(branch_filter)s) AND r.status = "active")
-        MATCH namespace_path = (object_permission)-[:HAS_ATTRIBUTE]->(:Attribute {name: "namespace"})-[:HAS_VALUE]->(object_permission_namespace:AttributeValue)
-            WHERE all(r IN relationships(namespace_path) WHERE (%(branch_filter)s) AND r.status = "active")
-        MATCH name_path = (object_permission)-[:HAS_ATTRIBUTE]->(:Attribute {name: "name"})-[:HAS_VALUE]->(object_permission_name:AttributeValue)
-            WHERE all(r IN relationships(name_path) WHERE (%(branch_filter)s) AND r.status = "active")
-        MATCH action_path = (object_permission)-[:HAS_ATTRIBUTE]->(:Attribute {name: "action"})-[:HAS_VALUE]->(object_permission_action:AttributeValue)
-            WHERE all(r IN relationships(action_path) WHERE (%(branch_filter)s) AND r.status = "active")
-        MATCH decision_path = (object_permission)-[:HAS_ATTRIBUTE]->(:Attribute {name: "decision"})-[:HAS_VALUE]->(object_permission_decision:AttributeValue)
-            WHERE all(r IN relationships(decision_path) WHERE (%(branch_filter)s) AND r.status = "active")
```

Comment:
these 4 MATCH queries have also been converted to subqueries. although these are different than those above because I think each `object_permission` should only have a single namespace, name, action, and decision
if each `object_permission` needs to support multiple namespaces, names, actions, and/or decisions, then the below subqueries will need to be changes to be more like those above